<?xml version="1.0" ?>
<launch>

  <!-- send table.xml to param server -->
  <arg name="bag" default="/home/zwu/LIDAR-DATA/3oct-around-pana1.bag" />
  <arg name="start_time" default="0" /> <!-- all timings are in seconds -->
  <arg name="play_duration" default="260" /> <!-- set negative to play all -->
  <arg name="namespace" default="" /> <!-- unused for now -->
  <arg name="resolution" default="1.2"/> 
  <arg name="step_size" default="0.05"/>
  <arg name="transformation_epsilon" default="0.0001"/>
  <arg name="max_iteration" default="100" />
  <arg name="voxel_leaf_size" default="2.0" />
  <arg name="min_scan_range" default="3.0" />
  <arg name="min_add_scan_shift" default="0.5" />  
  <arg name="min_add_scan_yaw_diff" default="0.013" />
  
  <!-- tf from lidar frame to car frame -->
  <arg name="tf_x" default="1.2"/>
  <arg name="tf_y" default="0.0"/>
  <arg name="tf_z" default="2.0"/>
  <arg name="tf_roll" default="0.0"/>
  <arg name="tf_pitch" default="0.0"/>
  <arg name="tf_yaw" default="0.0"/>

<!-- rosrun ndt_localizer ndt_mapping  -->
  <node pkg="ndt_mapping" type="large_scale_ndt_mapping" name="large_scale_ndt_mapping_$(arg namespace)" output="screen">
  	<param name="bag_file" value="$(arg bag)" />
  	<param name="start_time" value="$(arg start_time)" />
  	<param name="play_duration" value="$(arg play_duration)" />
    <param name="namespace" value="$(arg namespace)" />

  	<param name="resolution" value="$(arg resolution)" />
  	<param name="step_size" value="$(arg step_size)" />
  	<param name="transformation_epsilon" value="$(arg transformation_epsilon)" />
  	<param name="max_iteration" value="$(arg max_iteration)" />
  	<param name="voxel_leaf_size" value="$(arg voxel_leaf_size)" />
  	<param name="min_scan_range" value="$(arg min_scan_range)" />
  	<param name="min_add_scan_shift" value="$(arg min_add_scan_shift)" />
    <param name="min_add_scan_yaw_diff" value="$(arg min_add_scan_yaw_diff)" />
  	
  	<param name="tf_x" value="$(arg tf_x)" />
  	<param name="tf_y" value="$(arg tf_y)" />
  	<param name="tf_z" value="$(arg tf_z)" />
  	<param name="tf_roll" value="$(arg tf_roll)" />
  	<param name="tf_pitch" value="$(arg tf_pitch)" />
  	<param name="tf_yaw" value="$(arg tf_yaw)" />
  </node>
  
</launch>
